presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayContains.java

```diff
@@ -78,13 +79,26 @@ public FunctionInfo specialize(Map<String, Type> types, int arity, TypeManager t
         Type type = types.get("T");
         TypeSignature valueType = type.getTypeSignature();
         TypeSignature arrayType = parameterizedTypeName(StandardTypes.ARRAY, valueType);
-        MethodHandle methodHandle = methodHandle(ArrayContains.class, "contains", Type.class, MethodHandle.class, Block.class, type.getJavaType());
+
+        MethodHandle methodHandle;
+        if (type.getJavaType() == void.class) {
+            methodHandle = METHOD_HANDLE_UNKNOWN;
+        }
+        else {
+            methodHandle = methodHandle(ArrayContains.class, "contains", Type.class, MethodHandle.class, Block.class, type.getJavaType());
+        }
+
         MethodHandle equalsHandle = functionRegistry.getScalarFunctionImplementation(internalOperator(OperatorType.EQUAL, BooleanType.BOOLEAN, ImmutableList.of(type, type))).getMethodHandle();
         Signature signature = new Signature(FUNCTION_NAME, SCALAR, RETURN_TYPE, arrayType, valueType);
 
         return new FunctionInfo(signature, getDescription(), isHidden(), methodHandle.bindTo(type).bindTo(equalsHandle), isDeterministic(), true, ImmutableList.of(false, false));
     }
 
+    public static Boolean arrayWithUnknownType(Type elementType, MethodHandle equals, Block arrayBlock)
```

Comment:
You need to make it nullable only in the case where the argument is the "unknown" type. Making it nullable means that the Presto engine will expect you to use the boxed primitives and will call your function with `null`. For the "unknown" type you would like to use `void` for the argument type, but that's not allowed in Java, so you have to use `Void` and then tell the engine that the argument is nullable